Separate Upload Button from Plugin Nav/Search Bar

chickenn00dle opened the pull request:
This PR separates the Upload button from the Navigation/Search bar at
the top of the Plugins page. It also moves the button to the left of the
header nav/search to draw attention to search functionality.

Previously the placement of the layout of this header buried the search
functionality. Moving the Upload button to the left here makes it easier
to identify both Search and Upload.

Before:
https://d.pr/i/qsbCPX

After:
https://d.pr/i/5f1mZg

Fixes #33061

bluefuton commented:
Thanks for the PR @chickenn00dle!

I've noticed that Jetpack sites have two buttons, which can feel a bit odd:

<img width="1099" alt="Screen Shot 2019-09-15 at 10 51 07" src="https://user-images.githubusercontent.com/17325/64923314-c455a180-d7a6-11e9-817b-84274ae35a09.png">

How about moving the buttons out of the header altogether, and having a row of buttons above the plugin listing?

chickenn00dle commented:
Hey @bluefuton 

Sorry it took so long to get to this.

I think this is a good idea. Before doing anything, how about something like this?

![plugins-move-buttons](https://user-images.githubusercontent.com/17905991/65294801-4d4f4d00-db9a-11e9-8710-f001d06e320d.png)

bluefuton commented:
That looks much better to me @chickenn00dle! Let's get some design feedback before we proceed - I see you've added the Design Review label already :)

chickenn00dle commented:
Thanks for the feedback @bluefuton and @sixhours!

I made these changes along with a slight modification for smaller screen widths to be in line with the button in theme showcase (remove text and only show icon)

<img width="1146" alt="plugins-buttons" src="https://user-images.githubusercontent.com/17905991/65584449-dc55ce00-dfbb-11e9-8c6b-9be0f16144e5.png">

and

<img width="343" alt="plugins-buttons_mobile" src="https://user-images.githubusercontent.com/17905991/65584470-e677cc80-dfbb-11e9-9e18-798bd38d2fc2.png">

Let me know what you think!

sixhours commented:
I'm seeing a visual oddity with this patch when I click on Manage Plugins. 

On production:

<img width="970" alt="Screen Shot 2019-09-25 at 3 23 34 PM" src="https://user-images.githubusercontent.com/2124984/65632790-8592c900-dfa8-11e9-8360-e26c65006c37.png">
With this patch:

chickenn00dle commented:
Good catch @sixhours!

Looks like the styling I added in my first commit was responsible for this bug. I removed this styling and it's looking good now:

<img width="1232" alt="Screen Shot 2019-09-26 at 10 41 13" src="https://user-images.githubusercontent.com/17905991/65651700-3a30f700-e04a-11e9-8df7-548033e909c1.png">

On that note, do you think it's worth submitting a new PR to move the buttons in the manage view from the header as we did in this PR?

sixhours commented:
> On that note, do you think it's worth submitting a new PR to move the buttons in the manage view from the header as we did in this PR?

Yes, I think that's probably a good call! 👍

sixhours commented:
Ok, reviewing this now, most of the Gridicons are broken:

<img width="1280" alt="Screen Shot 2019-09-26 at 12 47 33 PM" src="https://user-images.githubusercontent.com/2124984/65708115-da434c00-e05b-11e9-9df6-53485c16eb2a.png">

I switched back to master and they're fine. The SVGs are 404ing with this patch, though:

<img width="1154" alt="Screen Shot 2019-09-26 at 12 44 45 PM" src="https://user-images.githubusercontent.com/2124984/65708123-de6f6980-e05b-11e9-9f92-b89c0a4cf99c.png">

I've dug into it a little but I can't see why the 404's. :-/

bluefuton commented:
Hey @sixhours! Looks like it was an unrelated gridicon issue that has since been fixed on `master`. I've rebased this branch and they're working for me now 👍

donpark commented:
Why do we have search bar above the buttons instead of the other way around?

Search bar dictates the content of the table below while buttons does not dictate nor depend on selection of the plugins.

chickenn00dle commented:
Hey @donpark !

That's a good point. The header component is not stuck to the top but looking through a few other views in Calypso, buttons below search seems to be a pattern.

For example the Theme Showcase:
<img width="1575" alt="Screen Shot 2019-09-27 at 12 07 10" src="https://user-images.githubusercontent.com/17905991/65742595-c404c100-e12b-11e9-8ca8-f55cc559b8e4.png">

Or media:
<img width="1563" alt="Screen Shot 2019-09-27 at 12 06 39" src="https://user-images.githubusercontent.com/17905991/65742607-cf57ec80-e12b-11e9-9916-85af1abc73f4.png">

May be worth opening a new issue for this?

chickenn00dle commented:
Thanks for the review! 

I'm actually not sure if I can go ahead and merge this on my own, or if I'd need someone else to do this for me?

@bluefuton @donpark
